.github/workflows/publish-brew-package.yml

```
@@ -0,0 +1,114 @@
+        required: true
+        type: string
+      HZ_DISTRIBUTION:
+        description: 'Distribution to be built: hazelcast or hazelcast-enterprise'
+        required: true
+        type: string
+      USE_TEST_REPO:
+        description: 'Use test repo for publishing'
+        required: true
+        type: string
+      PACKAGE_VERSION:
+        description: 'Version of the package e.g. 5.1.1, 5.1.1-1, defaults to HZ_VERSION'
+        type: string
+
+concurrency:
+  group: 'brew-${{ github.job }}-${{ inputs.HZ_VERSION }}-${{ inputs.HZ_DISTRIBUTION}}'
+
+jobs:
+  homebrew:
+    runs-on: macos-latest
+    env:
+      HZ_VERSION: ${{ inputs.HZ_VERSION }}
+      PACKAGE_VERSION: ${{ inputs.PACKAGE_VERSION || inputs.HZ_VERSION }}
+      HZ_DISTRIBUTION: ${{ inputs.HZ_DISTRIBUTION }}
+      PUBLISH: "true"
+      USE_TEST_REPO: ${{ inputs.USE_TEST_REPO }}
+    steps:
+      - name: Checkout hazelcast-packaging repo
+        uses: actions/checkout@v4
+
+      - name: Load env vars from .env file
+        run: cat .env >> $GITHUB_ENV
+
+      - name: Install up-to-date tools
+        run: |
+          brew install gnu-sed
+          brew install coreutils
+
+          PATH="$HOMEBREW_PREFIX/opt/gnu-sed/libexec/gnubin:$PATH"
+          sed --version
+          PATH="$HOMEBREW_PREFIX/opt/coreutils/libexec/gnubin:$PATH"
+          sha256sum --version
+          echo "PATH=$HOMEBREW_PREFIX/opt/coreutils/libexec/gnubin:$HOMEBREW_PREFIX/opt/gnu-sed/libexec/gnubin:$PATH" >> $GITHUB_ENV
+
+      - name: Setup JDK
+        uses: actions/setup-java@v4
+        with:
+          java-version: ${{ env.JAVA_VERSION }}
+          distribution: ${{ env.JAVA_DISTRIBUTION }}
+
+      - name: Download the distribution tar.gz file
+        run: |
+          HZ_PACKAGE_URL=$(mvn --batch-mode dependency:copy -Dartifact=com.hazelcast:${HZ_DISTRIBUTION}-distribution:${HZ_VERSION}:tar.gz -DoutputDirectory=./ -Dmdep.useBaseVersion=true | grep 'Downloaded from' | grep -Eo "https://[^ >]+${HZ_DISTRIBUTION}-distribution-.*.tar.gz")
+          echo "HZ_PACKAGE_URL=$HZ_PACKAGE_URL" >> $GITHUB_ENV
+
+      - name: Get homebrew repository
+        run: |
+          source ./common.sh
+          echo "BREW_GIT_REPO_NAME=${BREW_GIT_REPO_NAME}" >> $GITHUB_ENV
+
+      - name: Checkout homebrew-hz repo
+        uses: actions/checkout@v4
+        with:
+          repository: ${{ env.BREW_GIT_REPO_NAME }}
+          ref: master
+          token: ${{ secrets.DEVOPS_SECRET }}
+          path: 'homebrew-hz'
+
+      - name: Change the artifact in homebrew-hz
+        run: |
+          ./build-hazelcast-homebrew-package.sh
+
+      - name: Commit changes & Push to homebrew-hz repo
+        run: |
+          source common.sh
+
+          cd homebrew-hz
+          git config --global user.name 'devOpsHazelcast'
+          git config --global user.email 'devops@hazelcast.com'
```

```suggestion
          git config --global user.name "${GITHUB_ACTOR}"
          git config --global user.email "${GITHUB_ACTOR}@users.noreply.github.com"
```

Nit - would be neater if this was the details of the user running the action